tools/make.py

```diff
@@ -598,28 +614,26 @@ def set_version_in_files():
                 f.close()
 
                 if fileText:
-                    # Search and save version stamp, search short if long not found
-                    versionFound = re.findall(pattern, fileText)
-                    if not versionFound:
-                        versionFound = re.findall(patternShort, fileText)
+                    # Version string files
+                    # Search and save version stamp
+                    versionsFound = re.findall(pattern, fileText) + re.findall(patternShort, fileText)
+                    # Filter out sub-versions of other versions
+                    versionsFound = [j for i, j in enumerate(versionsFound) if all(j not in k for k in versionsFound[i + 1:])]
 
                     # Replace version stamp if any of the new version parts is higher than the one found
-                    if versionFound:
-                        # First item in the list findall returns
-                        versionFound = versionFound[0]
-
-                        newVersionUsed = ""
-                        # Use the same version length as the one found
-                        if len(versionFound) == len(newVersion):
-                            newVersionUsed = newVersion
-                        if len(versionFound) == len(newVersionShort):
-                            newVersionUsed = newVersionShort
-
-                        # Print change and modify the file if changed
-                        if versionFound != newVersionUsed:
-                            print_green("Changing version {} => {} in {}".format(versionFound, newVersionUsed, filePath))
-                            replace_file(filePath, versionFound, newVersionUsed)
-
+                    for versionFound in versionsFound:
+                        if versionFound:
+                            # Use the same version length as the one found
+                            newVersionUsed = "" # In case undefined
+                            if len(versionFound) == len(newVersion):
+                                newVersionUsed = newVersion
+                            if len(versionFound) == len(newVersionShort):
```

Comment:
I think this will fail when switching from 3.9 to 3.10 as the string length changes
This might work better:
```
if versionFound.count(".") == newVersion.count("."):
    newVersionUsed = newVersion
if versionFound.count(".") == newVersionShort.count("."):
    newVersionUsed = newVersionShort
```